core/chains/evm/gas/universal_estimator.go

```diff
@@ -0,0 +1,412 @@
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas/rollups"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+// metrics are thread safe
+var (
+	promUniversalEstimatorGasPrice = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "gas_price_updater",
+		Help: "Sets latest gas price (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorBaseFee = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "base_fee_updater",
+		Help: "Sets latest BaseFee (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorMaxPriorityFeePerGas = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "max_priority_fee_per_gas_updater",
+		Help: "Sets latest MaxPriorityFeePerGas (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+	promUniversalEstimatorMaxFeePerGas = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "max_fee_per_gas_updater",
+		Help: "Sets latest MaxFeePerGas (in Wei)",
+	},
+		[]string{"evmChainID"},
+	)
+)
+
+const (
+	MinimumBumpPercentage   = 10 // based on geth's spec
+	ConnectivityPercentile  = 80
+	BaseFeeBufferPercentage = 40
+)
+
+type UniversalEstimatorConfig struct {
+	BumpPercent  uint16
+	CacheTimeout time.Duration
+	EIP1559      bool
+
+	BlockHistorySize uint64
+	RewardPercentile float64
+	HasMempool       bool
+}
+
+type universalEstimatorClient interface {
+	SuggestGasPrice(ctx context.Context) (*big.Int, error)
+	FeeHistory(ctx context.Context, blockCount uint64, rewardPercentiles []float64) (feeHistory *ethereum.FeeHistory, err error)
+}
+
+type UniversalEstimator struct {
+	services.StateMachine
+
+	client  universalEstimatorClient
+	logger  logger.Logger
+	config  UniversalEstimatorConfig
+	chainID *big.Int
+
+	gasPriceMu sync.RWMutex
+	gasPrice   *assets.Wei
+
+	dynamicPriceMu sync.RWMutex
+	dynamicPrice   DynamicFee
+
+	priorityFeeThresholdMu sync.RWMutex
+	priorityFeeThreshold   *assets.Wei
+
+	l1Oracle rollups.L1Oracle
+
+	wg     *sync.WaitGroup
+	stopCh services.StopChan
+}
+
+func NewUniversalEstimator(lggr logger.Logger, client universalEstimatorClient, cfg UniversalEstimatorConfig, chainID *big.Int, l1Oracle rollups.L1Oracle) *UniversalEstimator {
+	return &UniversalEstimator{
+		client:   client,
+		logger:   logger.Named(lggr, "UniversalEstimator"),
+		config:   cfg,
+		chainID:  chainID,
+		l1Oracle: l1Oracle,
+		wg:       new(sync.WaitGroup),
+		stopCh:   make(chan struct{}),
+	}
+}
+
+func (u *UniversalEstimator) Start(context.Context) error {
+	return u.StartOnce("UniversalEstimator", func() error {
+		if u.config.BumpPercent < MinimumBumpPercentage {
+			return fmt.Errorf("BumpPercent: %s is less than minimum allowed percentage: %s",
+				strconv.FormatUint(uint64(u.config.BumpPercent), 10), strconv.Itoa(MinimumBumpPercentage))
+		}
+		if u.config.EIP1559 && u.config.RewardPercentile > ConnectivityPercentile {
+			return fmt.Errorf("RewardPercentile: %s is greater than maximum allowed connectivity percentage: %s",
```

Comment:
I think the message would require too much context to understand with `connectivity percentage`. Maybe we can keep it even simpler like
```suggestion
			return fmt.Errorf("RewardPercentile: %s is greater than maximum allowed percentile: %s",
```